dfaligner/model.py

```diff
@@ -81,6 +84,29 @@ def forward(self, x):
         x = self.lin(x)
         return x
 
+    def on_load_checkpoint(self, checkpoint):
+        """Deserialize the checkpoint hyperparameters.
+        Note, this shouldn't fail on different versions of pydantic anymore,
+        but it will fail on breaking changes to the config. We should catch those exceptions
+        and handle them appropriately."""
+        self.config = AlignerConfig(**checkpoint["hyper_parameters"]["config"])
+
+    def on_save_checkpoint(self, checkpoint):
+        """Serialize the checkpoint hyperparameters"""
+        checkpoint["hyper_parameters"]["config"] = self.config.model_dump(
+            mode="json",
+            exclude={
+                "path_to_preprocessing_config_file": True,
+                "path_to_text_config_file": True,
+                "path_to_audio_config_file": True,
+                "path_to_training_config_file": True,
+                "path_to_model_config_file": True,
+                "path_to_aligner_config_file": True,
+                "path_to_vocoder_config_file": True,
+                "path_to_feature_prediction_config_file": True,
+            },
```

Comment:
These lines occur in three submodules, and in each case it's for a subclass of `PartialLoadConfig`. Could we have a method `PartialLoadConfig.paths_to_config_file_fields()` that would use introspection to find all fields that match `path_to_*_config_file` and return this exclude dict, so that here we could write `exclude=self.paths_to_config_file_fields()` here and in the other submodules?

Reply:
Or else, since it's the same list in each case, we could be lazier and just hard-code the list in one place, that would be fine too, as long as we don't forget to augment it if we add more path-to-config-file fields later.

Reply:
Wow, I was confused about the type for the `exclude` argument, i.e., why a dict when the `model_dump()` documentation says it should be a list, and I figured it out: the type for `include` and `exclude` is `IncEx` which is defined as `set[int] | set[str] | dict[int, _IncEx] | dict[str, _IncEx] | None`. So a dict or a set work, but surprisingly not an actual list even though the API documentation says "A list of fields to exclude from the output." 😄